File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Domain/Enrollment/Events/EnrollmentRequested.cs
using CreditCardEnrollment.Common.Events;

namespace CreditCardEnrollment.Domain.Enrollment.Events;

public class EnrollmentRequested : CreationEvent
{
    public string UserId { get; set; } = string.Empty;
    public string ProductId { get; set; } = string.Empty;
    public int AnnualIncomeInCents { get; set; }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/creditCard/enrollment/event/EnrollmentRequested.ts
import { CreationEvent } from '../../../common/event/CreationEvent';
import { Enrollment } from '../aggregate/Enrollment';
import { EnrollmentStatus } from '../aggregate/EnrollmentStatus';

export class EnrollmentRequested extends CreationEvent<Enrollment> {
    constructor(
        eventId: string,
        aggregateId: string,
        aggregateVersion: number,
        correlationId: string,
        causationId: string,
        recordedOn: Date,
        public readonly userId: string,
        public readonly productId: string,
        public readonly annualIncomeInCents: number
    ) {
        super(eventId, aggregateId, aggregateVersion, correlationId, causationId, recordedOn);
    }

    createAggregate(): Enrollment {
        return new Enrollment(
            this.aggregateId,
            this.aggregateVersion,
            this.userId,
            this.productId,
            EnrollmentStatus.Requested,
            this.annualIncomeInCents,
            this.recordedOn
        );
    }
}